src/components/Stories/index.js

```diff
@@ -1,62 +1,35 @@
 import PropTypes from "prop-types";
-import React, { useEffect, useState } from "react";
-import useSWR from "swr";
+import React from "react";
 
 import List from "./List";
 import useStyles from "./useStyles";
 
 import FeaturedStoryCard from "@/pesayetu/components/FeaturedStoryCard";
 import Loading from "@/pesayetu/components/Loading";
 import Pagination from "@/pesayetu/components/Pagination";
-import fetchAPI from "@/pesayetu/utils/fetchApi";
 
 function Stories({
   featuredStoryProps,
   items: itemsProp,
   category,
   pagination,
   page,
-  paginate,
+  onPaginate,
+  isLoading,
   ...props
 }) {
   const classes = useStyles(props);
   const variant = category === "insights" ? "embed" : undefined;
 
-  const [stories, setStories] = useState(itemsProp);
-  const [isLoading, setIsLoading] = useState(false);
-
   const handleClickPage = (_, value) => {
-    if (paginate) {
-      paginate(value);
+    if (onPaginate) {
+      onPaginate(value);
     }
   };
 
-  const { data, error } = useSWR("/api/wp/archive", (url) => {
-    let offset;
-    if (page === 1) {
-      offset = 0;
-    } else if (page === 2) {
-      offset = 6;
-    } else {
-      offset = (page - 2) * 9 + 6;
-    }
-    return fetchAPI(`${url}/?taxonomyId=${category}&offset=${offset}`);
-  });
-
-  useEffect(() => {
-    if (!data && !error) {
-      setIsLoading(true);
-    } else {
-      if (data) {
-        setStories(data);
-      }
-      setIsLoading(false);
-    }
-  }, [data, error]);
-
-  let items = stories;
+  let items = itemsProp;
   if (page === 1) {
-    items = stories
+    items = itemsProp
       ?.filter(({ slug }) => slug !== featuredStoryProps?.slug)
       ?.slice(0, 6);
   }
```

Comment:
```suggestion
    items = itemsProp
      ?.filter(({ slug }) => slug !== featuredStoryProps?.slug)
      ?.slice(0, 6);
  }
  
  const total = pagination?.offsetPagination?.total ?? 0;
  const count = total ? 1 + Math.ceil(Math.max(total - 6, 0) / 9) : 0
```

And use `count` in `Pagination`... otherwise, we're off by at least 1 page.